src/dmd/builtin.d

```diff
@@ -103,6 +103,130 @@ extern (C++) Expression eval_ldexp(Loc loc, FuncDeclaration fd, Expressions* arg
     return new RealExp(loc, CTFloat.ldexp(arg0.toReal(), cast(int) arg1.toInteger()), arg0.type);
 }
 
+extern (C++) Expression eval_log(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.log(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_log2(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.log2(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_log10(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.log10(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_expm1(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.expm1(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_exp2(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.exp2(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_nearbyint(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.nearbyint(arg0.toReal()), arg0.type);
+}
+
+extern (C++) Expression eval_rint(Loc loc, FuncDeclaration fd, Expressions* arguments)
+{
+    Expression arg0 = (*arguments)[0];
+    assert(arg0.op == TOK.float64);
+    return new RealExp(loc, CTFloat.rint(arg0.toReal()), arg0.type);
```

Comment:
Likewise having this as a built-in too.

Reply:
These two are there for compatibility with LDC, and also as a simple matter of convenience so the user doesn't have to use two paths for the same code.

Reply:
Well I'd rather not add builtins that aren't common for all, as it means you will _have to_ use two paths anyway.  For example y2lx comes with version(INLINE_Y2LX).

Reply:
What's the issue with adding them to gdc?

Reply:
These are non existent builtins in gcc.

More generally though adding them as builtins would make them more susceptible to wrong code in the const folder/optimizer given.
```
setRoundingMode(down);
immutable x = rint(1.8);
setRoundingMode(up);
immutable y = rint(1.8);
```